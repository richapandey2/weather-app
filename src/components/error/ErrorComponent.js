import React from "react";
import { Box, Button } from "@material-ui/core";
import PropTypes from "prop-types";
import { Alert, AlertTitle } from "@material-ui/lab";
import {
  ERROR_MESSAGE_LABEL,
  ERROR_LABEL_BUTTON_LABEL,
  ERROR_TITLE,
} from "../../constants/Constants";

function ErrorComponent({ fetchWeatherData }) {
  return (
    <Box>
      <Alert severity="error">
        <AlertTitle>{ERROR_TITLE}</AlertTitle>
        {ERROR_MESSAGE_LABEL}
        <Button
          variable="contained"
          color="secondary"
          onClick={fetchWeatherData}
        >
          {ERROR_LABEL_BUTTON_LABEL}
        </Button>
      </Alert>
    </Box>
  );
}

ErrorComponent.propTypes = {
  fetchWeatherData: PropTypes.func,
};

export default ErrorComponent;
