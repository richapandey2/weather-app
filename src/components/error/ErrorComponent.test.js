import React from "react";
import { render, cleanup } from "@testing-library/react";
import ErrorComponent from "./ErrorComponent";

afterEach(cleanup);

test("should test error component", () => {
  const { asFragment } = render(<ErrorComponent />);
  expect(asFragment).toMatchSnapshot();
});
