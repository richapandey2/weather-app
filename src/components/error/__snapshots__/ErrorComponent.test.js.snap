// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test error component 1`] = `[Function]`;
